src/lib.rs

```diff
@@ -1680,6 +1744,17 @@ impl DnsOutgoing {
             return;
         }
 
+        for sub in &service.sub_domain {
+            debug!("Adding subdomain {}", sub);
+            self.add_additional_answer(Box::new(DnsPointer::new(
+                &sub,
+                TYPE_PTR,
+                CLASS_IN,
+                service.other_ttl,
+                service.fullname.clone(),
+            )));
+        }
```

Comment:
Removing it does not seem to break anything on dns-sd or the test, so I guess you are right.

Reply:
I think we probably would need to send the sub_domain PTR when the query is for the sub_domain.  You didn't see any problems in tests, maybe because the initial broadcast of the server are cached, or something I didn't know. 

So, instead of removing the sub_domain PTR send, how about making it conditional? 

1. Add a new param like `is_sub: bool` to `add_answer_with_additionals`. 

inside the method,  we will send sub_domain PTR if `is_usb` is true. (the regular PTR is always sent).

3. At the calling place `handle_query`,  we can have different branches like this (using the check you have):  

```
               if question.entry.name == service.ty_domain {
                        out.add_answer_with_additionals(&msg, service, false);
               } else if service
                             .sub_domain
                             .as_ref()
                             .map_or(false, |v| v == &question.entry.name) {
                        out.add_answer_with_additionals(&msg, service, true);
               } else ....
```

I'm also OK if you think the above is complicated and you want to add the earlier code back.

Reply:
It feels a bit complicated indeed :)